spring-rabbit/src/main/java/org/springframework/amqp/rabbit/AsyncRabbitTemplate.java

```diff
@@ -713,8 +722,10 @@ void setNackCause(String nackCause) {
 
 		void startTimer() {
 			if (AsyncRabbitTemplate.this.receiveTimeout > 0) {
-				this.timeoutTask = AsyncRabbitTemplate.this.taskScheduler.schedule(new TimeoutTask(), // NOSONAR sync
-						new Date(System.currentTimeMillis() + AsyncRabbitTemplate.this.receiveTimeout));
+				synchronized (AsyncRabbitTemplate.this.taskScheduler) {
```

Comment:
![image](https://cloud.githubusercontent.com/assets/1071590/21828575/34817236-d75f-11e6-90bc-c610c78f02c1.png)

Not sure if that is so big deal, maybe something like `final Object taskSchedulerMonitor` would be better any way ?
At least, let's add some comment here to avoid similar discussions in the future